system/jlib/jfile.hpp

```diff
@@ -632,4 +633,67 @@ const static bool filenamesAreCaseSensitive = false;
 const static bool filenamesAreCaseSensitive = true;
 #endif
 
+class jlib_decl CSortedDirectoryIterator : implements IDirectoryIterator, public CInterface
```

Comment:
Better to use CSimpleInterface unless you need beforeDispose
And better to use templaces to avoid double inheritance.
so:
public CSimpleInterfaceOf<IDirectoryIterator>